### PR TITLE
io.github.aerocyber.sitemarker

### PR DESCRIPTION
**Pull request by aerocyber**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by aerocyber** [comment]
bot, build io.github.aerocyber.sitemarker

File: io.github.aerocyber.sitemarker.svg
Comment (hfiguiere) [inline io.github.aerocyber.sitemarker.svg:1]: this file is upstream it shouldn't be here. it's not even referenced.

File: io.github.aerocyber.sitemarker.desktop
Comment (hfiguiere) [inline io.github.aerocyber.sitemarker.desktop:1]: this file is upstream it shouldn't be here it's not even referenced.

File: io.github.aerocyber.sitemarker.appdata.xml
Comment (hfiguiere) [inline io.github.aerocyber.sitemarker.appdata.xml:1]: this file is upstream it shouldn't be here it's not even referenced.

File: io.github.aerocyber.sitemarker.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <keyword>bookmark viewer</keyword>
+    <keyword>bookmark viewers</keyword>
+    <keyword>viewer</keyword>
+    <keyword>viewers</keyword>
+    <keyword>open source</keyword>
+    <keyword>mit</keyword>
+    <keyword>MIT</keyword>
+    <keyword>free</keyword>
+  </keywords>
+</component>
```
Comment (hfiguiere) [inline io.github.aerocyber.sitemarker.appdata.xml:52]: mandatory `releases` (at least one) are missing

File: io.github.aerocyber.sitemarker.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "sitemarker",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+	"--filesystem=home",
+        "--filesystem=~/.sitemarker:create"
```
Comment (hfiguiere) [inline io.github.aerocyber.sitemarker.json:14]: there is redundancy here. Also do you really need home? Doesn't the app use standard location for directories? (isn't it a Gtk app?)

**Comment by hfiguiere** [comment]
build fails because the icon isn't square.

File: io.github.aerocyber.sitemarker.json (hunk 2 above)
Comment (aerocyber) [reply io.github.aerocyber.sitemarker.json:14]: Do you mean that only `--filesystem=~/.sitemarker:create" is needed without `--filesystem=home`? Just making sure.

Comment (hfiguiere) [reply io.github.aerocyber.sitemarker.json:14]: so it probably use the standard XDG location, and portal, which mean that these permissions are excessive.

Comment (aerocyber) [reply io.github.aerocyber.sitemarker.json:14]: I'm sorry, I should have mentioned earlier that the app has a hardcoded path to `~/.sitemarker`. So, would removing `--filesystem=home` be enough or should I change the upstream?

**Comment by aerocyber** [comment]
bot, build io.github.aerocyber.sitemarker

**Comment by aerocyber** [comment]
bot, build io.github.aerocyber.sitemarker

**Comment by aerocyber** [comment]
The previous build had issues regarding displaying keyboard shortcuts. The new build is initialized for testing them

**Comment by aerocyber** [comment]
bot, build io.github.aerocyber.sitemarker

File: io.github.aerocyber.sitemarker.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "runtime-version" : "44",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "sitemarker",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=~/.sitemarker:create"
```
Comment (barthalion) [inline io.github.aerocyber.sitemarker.json:13]: To follow-up on the discussion from the old thread: you should update your app to respect XDG_CONFIG_HOME variable. This way you will be able to drop this line altogether.

Comment (aerocyber) [reply io.github.aerocyber.sitemarker.json:13]: Alright, I will do that

**Comment by aerocyber** [comment]
This build was found to work correctly after the suggested changes.

**Comment by aerocyber** [comment]
Thank you, for your time and help!

**Pull request closed without merging**